pkg/ride/diff_state.go

```diff
@@ -253,20 +253,24 @@ func (ds *diffState) burnNewAsset(assetID crypto.Digest, quantity int64) {
 }
 
 func (ds *diffState) lease(sender, receiver proto.AddressID, amount int64) error {
-	if senderDiff, ok := ds.wavesBalances[sender]; ok {
-		err := senderDiff.addLeaseOut(amount) // Increase sender's leaseOut by leasing amount
-		if err != nil {
-			return err
-		}
-		ds.wavesBalances[sender] = senderDiff
+	senderDiff, err := ds.loadWavesBalance(sender)
```

Comment:
the call of `loadWavesBalance` is absolutely unnecessary, because it was called just before `lease` function was called